ViewEnvironmentUI/Sources/ViewEnvironmentPropagating.swift

```diff
@@ -119,6 +119,10 @@ extension ViewEnvironmentPropagating {
     public var environment: ViewEnvironment {
```

Comment:
more of a general question about the changeset – is this alternate API mainly for convenience? curious what the motivating examples are (if they exist).

Reply:
It's mainly for convenience. The motivating example is support for what we call "pane-based size class", where a split container VC is able to override the horizontal size class for a particular pane based on the available horizontal width of that pane.

Without this change we'd either need to insert a node between the split container VC and it's children, or introduce a wrapper VC around those children to perform the environment mutation.